managed/services/dbaas/kubernetes/types.go

```diff
@@ -0,0 +1,336 @@
+	gCacheSize := "600M"
+	if cluster.Params.Pxc.Configuration == "" {
+		if memory > memorySmallSize && memory <= memoryMediumSize {
+			gCacheSize = "2.4G"
+		}
+		if memory > memoryMediumSize && memory <= memoryLargeSize {
+			gCacheSize = "9.6G"
+		}
+		if memory > memoryLargeSize {
+			gCacheSize = "9.6G"
+		}
+		cluster.Params.Pxc.Configuration = fmt.Sprintf(pxcDefaultConfigurationTemplate, gCacheSize)
+	}
+	dbCluster := &dbaasv1.DatabaseCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cluster.Name,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: dbaasAPI,
+			Kind:       dbaasKind,
+		},
+		Spec: dbaasv1.DatabaseSpec{
+			Database:       databasePXC,
+			DatabaseImage:  cluster.Params.Pxc.Image,
+			DatabaseConfig: cluster.Params.Pxc.Configuration,
+			ClusterSize:    cluster.Params.ClusterSize,
+			DBInstance: dbaasv1.DBInstanceSpec{
+				DiskSize: strconv.FormatInt(cluster.Params.Pxc.DiskSize, 10),
+				CPU:      fmt.Sprintf("%dm", cluster.Params.Pxc.ComputeResources.CpuM),
+				Memory:   strconv.FormatInt(cluster.Params.Pxc.ComputeResources.MemoryBytes, 10),
+			},
+			Monitoring: dbaasv1.MonitoringSpec{
+				PMM: &dbaasv1.PMMSpec{},
+			},
+			LoadBalancer: dbaasv1.LoadBalancerSpec{},
+			Backup:       dbaasv1.BackupSpec{},
+		},
+	}
+	if cluster.Params.Pxc.StorageClass != "" {
+		dbCluster.Spec.DBInstance.StorageClassName = &cluster.Params.Pxc.StorageClass
+	}
+	if cluster.Params.Haproxy != nil {
+		resources, err := convertComputeResource(cluster.Params.Haproxy.ComputeResources)
+		if err != nil {
+			return nil, err
+		}
+		dbCluster.Spec.LoadBalancer.Image = cluster.Params.Haproxy.Image
+		dbCluster.Spec.LoadBalancer.Size = cluster.Params.ClusterSize
+		dbCluster.Spec.LoadBalancer.Resources = resources
+		dbCluster.Spec.LoadBalancer.Type = "haproxy"
+	}
+	if cluster.Params.Proxysql != nil {
+		resources, err := convertComputeResource(cluster.Params.Proxysql.ComputeResources)
+		if err != nil {
+			return nil, err
+		}
+		dbCluster.Spec.LoadBalancer.Size = cluster.Params.ClusterSize
+		dbCluster.Spec.LoadBalancer.Image = cluster.Params.Proxysql.Image
+		dbCluster.Spec.LoadBalancer.Resources = resources
+		dbCluster.Spec.LoadBalancer.Type = "proxysql"
+	}
+	if cluster.Expose {
+		exposeType, ok := exposeTypeMap[clusterType]
+		if !ok {
+			return dbCluster, fmt.Errorf("failed to recognize expose type for %s cluster type", clusterType)
+		}
+		dbCluster.Spec.LoadBalancer.ExposeType = exposeType
+		annotations, ok := exposeAnnotationsMap[clusterType]
+		if !ok {
+			return dbCluster, fmt.Errorf("failed to recognize expose annotations for %s cluster type", clusterType)
+		}
+		dbCluster.Spec.LoadBalancer.Annotations = annotations
+		if cluster.InternetFacing && clusterType == ClusterTypeEKS {
+			dbCluster.Spec.LoadBalancer.Annotations["service.beta.kubernetes.io/aws-load-balancer-type"] = "external"
+		}
+
+	}
+	var sourceRanges []string
+	for _, sourceRange := range cluster.SourceRanges {
```

Comment:
In the future we should validate the ranges.

Reply:
Agree.